APIKit/RequestBodyBuilder.swift

```diff
@@ -24,26 +19,19 @@ public enum RequestBodyBuilder {
         }
     }
 
-    public func buildBodyFromObject(object: AnyObject) -> Result<NSData, NSError> {
+    public func buildBodyFromObject(object: AnyObject) throws -> NSData {
         switch self {
         case .JSON(let writingOptions):
-            if !NSJSONSerialization.isValidJSONObject(object) {
-                let userInfo = [NSLocalizedDescriptionKey: "invalid object for JSON passed."]
-                let error = NSError(domain: APIKitRequestBodyBuidlerErrorDomain, code: 0, userInfo: userInfo)
-                return .failure(error)
-            }
-
-            return try { error in
-                return NSJSONSerialization.dataWithJSONObject(object, options: writingOptions, error: error)
+            guard NSJSONSerialization.isValidJSONObject(object) else {
+                throw NSError(domain: NSCocoaErrorDomain, code: 3840, userInfo: nil)
```

Comment:
If `isValidJSONObject(_:)` is false, `dataWithJSONObject(_:options:)` throws `NSException`. So, if we omit `isValidJSONObject(_:)`, the app crashes when invalid object passed to request body builder. I'll add comment about this.